src/main/jbake/content/docs/components/metrics/index.adoc

```diff
@@ -4,22 +4,181 @@ Thomas Segismont
+toc::[]
 
+== Introduction
+Hawkular Metrics is a metrics storage engine built on top of link:http://cassandra.apache.org[Cassandra] that offers
+a rich, growing set of features that includes.
 
-Three types of metrics are currently supported:
+* Multi tenancy
+* Multiple metric types
+* Tagging
+* Automatic data removal
+* REST API
+* Scalability
+* Rich query support
 
- * Availability: metric that represent availability: up or down
- * Counter: numeric value that always goes up; example: the total number of visitors of a web page
-   ** Rates: rate of change can be retrieve for counter rates
- * Gauge: numeric value that can go up or down; example: the response time on an application
-   ** Aggregate buckets (min/max/average on some intervals) can be obtained for gauges when individual datapoints are not needed
+== Identifiers
+All identifiers are stored as variable length, UTF-8 encoded strings. This includes,
 
-Tags are available for easy filtering and querying, along with retention times for automatic pruning of older data.
+* Tenant ids
+* Metric names
+* Tag keys (for both metric and data point tags)
 
+[TIP]
+At present there is no restriction on characters that can be used in identifiers. This may change in the future
+though (See https://issues.jboss.org/browse/HWKMETRICS-208[HWKMETRICS-208] for details). For this reason it is
+recommended to restrict the characters to letters, numbers, underscore, period, and forward slash.
 
+== Multi Tenancy
+Hawkular Metrics provides _virtual_ multi tenancy. All data is mapped to a tenant, but the data on disk is not
+physically partitioned by tenant. From an API perspective though, everything is partitioned by tenant. All requests,
+both reads and writes, must include a tenant id.
+
+A tenant has an id that uniquely identifies it. The id is a variable length, UTF-8 encoded string.
+
+TIP: Tenants do not have to be explicitly created. They will be created when inserting data points.
+
+== Metric Types
+A metric represents a single time series or continuous stream of data points. A data point consists of a timestamp and
+a value. The type of value will vary depending on the metric type. Three types of metrics are currently supported:
+
+* Availability
+* Gauge
+* Counter
+
+TIP: Metrics do not have to be explicitly created. They will be created when inserting data points.
+
+TIP: A string metric type may be added in a future release. See
+https://issues.jboss.org/browse/HWKMETRICS-384(HWKMETRICS-384) for details.
+
+=== Availability
+Represents the availability of a resource such as host machine (physical or virtual) or an application server. There
+are only three supported availability types or values,
+
+* up
+* down
+* unknown
+
+Availability is stored as single, unsigned byte.
+
+=== Gauge
+Has a numeric value that can fluctuate, going up or down. Some examples of gauges include,
```

Comment:
Also, gauge is different from counters in a way that it's not necessarily related to previous values in any way. Amount of people at point X in time does not necessarily correlate to amount of people at point Y in time.